setup.sh

```diff
@@ -134,6 +134,28 @@ if [[ $DEVICE == "tpu" ]]; then
     fi
 fi
 
+EXTRA_PIP_INSTALL_ARGS=()
+if [ "$MODE" != 'stable' ]; then
+  EXTRA_PIP_INSTALL_ARGS+=(--no-deps)
+fi
+
+if [ -f 'requirements-pyproj.txt' ]; then
+  python3 -m uv pip install -r requirements-pyproj.txt
+elif [ -f "${MAXTEXT_REPO_ROOT}"'/requirements-pyproj.txt' ]; then
+  python3 -m uv pip install -r "${MAXTEXT_REPO_ROOT}"'/requirements-pyproj.txt'
+else
+  >&2 printf 'Could not find requirements-pyproj.txt, looked in ("%s", "%s")\n' \
+      "${PWD:-$(pwd)}" \
+      "${MAXTEXT_REPO_ROOT}"
+  exit 2
+fi
+
+# Install maxtext package
+if [ -f 'pyproject.toml' ]; then
+  python3 -m uv pip install -e . "${EXTRA_PIP_INSTALL_ARGS[@]}" --resolution=lowest
+  install_maxtext_github_deps
+fi
```

Comment:
Can you confirm this will still work well when user manually passes in `JAX_VERSION`? My understanding is that everything will install normally, then just override JAX/libtpu/etc. versions at the end, right?